doc/dsdr/0010-placement-modelling.md

```diff
@@ -0,0 +1,105 @@
+
+Having found a candidate match, we then need to take our match and combine it with the head of the placement sequence. This will extend the existing placement sequence by exactly the length of the matched segment (6 years, or less for an offset segment or one which takes the child past 18 years of age). We need to decide if we are going to loop further. If we have a terminal segment or if the child has exceeded 18 years of age then we stop.
+
+If we don't stop, then the loop continues with the extended placement sequence created by adding this segment to previous head of the placement sequence as the new sequence to be matched.
+
+**Rejection sampling**
+
+The process described above is designed to create a large number of placement pathways which are similar to those observed historically but are nonetheless different in terms of the exact age at which children join, the exact age at which they leave, and the exact duration of the placement sequence. Novel placement sequences can be created, but only by chaining 6-year subsections (with initial shorter offset segments) together. This clearly constrains the amount of novel placement sequences which can be created. However, there are lots of subtly different lengths created. This is important, because it's what allows rejection sampling to work.
+
+Rejection sampling is a process of making one distribution like another through selective resampling. For example, if I had a bag of 10 blue and 5 red marbles but wanted to sample (with replacement) a uniform distribution of blue and red I could employ rejection sampling after each sample. Every time I draw a blue marble from the bag I could sample a random number between 0 and 1. If the number is greater than 10/5, or 0.5 (the rejection threshold), I would log a blue sample and return the marble. If it is less than 0.5 I would return the marble without logging the sample and repeat the process. For red marbles the rejection threshold would be 0. If I repeated this process lots of times, the accepted samples would show a distribution equivalent to if I had an even number of blue and red marbles in the bag - a uniform distribution. Careful calibration of the rejection thresholds (0.5 and 0 in this example) for each sampled value can theoretically turn any candidate distribution into any other target distribution, provided the candidate distribution has some probability density everywhere the target distribution has probability density. This is because rejection sampling can only reject proposed samples, it can't create samples if they're not generated in thef irst place. This is equivalent to saying that we can't create a uniform reb-blue-green marble distribution if there aren't any green marbles in the bag.
```

Comment:
Is there a good description of rejection sampling we could link to? A formal and/or an informal one?

Reply:
tho, this is a good explanation and the first time I properly understand how the rejection sampling workd.